src/main/java/edu/illinois/library/cantaloupe/resource/AbstractResource.java

```diff
@@ -98,7 +98,7 @@ protected static Reference getPublicReference(Reference requestRootRef,
         if (".".equals(appRootRelativePath)) { // when the paths are the same
             appRootRelativePath = "";
         }
-        Reference newRef;
+        Reference newRef = requestRef;
```

Comment:
Could you do `Reference newRef = new Reference(requestRef);` here to avoid modifying `requestRef`?